src/window/index.ts

```diff
@@ -23,15 +23,11 @@ function getWebRoot() {
 }
 
 /**
- * Open a given window; if it is already open, focus it.
- * @param name The window identifier; this controls window re-use.
- * @param url The URL to load into the window.
- * @param options A hash of options used by `new BrowserWindow(options)`
- * @param prefs Options to control the new window.
+ * Restore or focus a window if it is already open
+ * @param window The Electron Browser window to show or restore
+ * @returns Boolean: True if the browser window is shown or restored
  */
-function createWindow(name: string, url: string, options: Electron.BrowserWindowConstructorOptions) {
-  let window = (name in windowMapping) ? BrowserWindow.fromId(windowMapping[name]) : null;
-
+export const didRestoreWindow = (window: Electron.BrowserWindow | null): window is Electron.BrowserWindow => {
```

Comment:
My original intent was to name this function in a way so that it would be clear that it was answering the question of "did this window restore/get focus? If not, we need to create a new window". I originally started with something like `hasWindowToRestore`, but that didn't make sense because we're actually performing an action; it was then a tossup between `canRestoreWindow` vs `didRestoreWindow`.

In hindsight, I guess prefixing with `did` kind of has that association with checking if something happened in the past.. I'm open to prefixing with `try`, but in my mind that would infer that a function has wrapped some other function in a `try...catch` block. Perhaps that's an incorrect assumption on my part.. in the end I'd rather go with whatever is most clear to the rest of the team.

Reply:
I think `restoreWindow` is a fair compromise. It honestly gets the point across quite effectively without any of the `try`, `did`, `can` nonsense 😆